pkg/controller/controller_utils_test.go

```diff
@@ -382,103 +410,172 @@ func TestDeletePodsAllowsMissing(t *testing.T) {
 
 func TestActivePodFiltering(t *testing.T) {
 	logger, _ := ktesting.NewTestContext(t)
-	// This rc is not needed by the test, only the newPodList to give the pods labels/a namespace.
-	rc := newReplicationController(0)
-	podList := newPodList(nil, 5, v1.PodRunning, rc)
-	podList.Items[0].Status.Phase = v1.PodSucceeded
-	podList.Items[1].Status.Phase = v1.PodFailed
-	expectedNames := sets.NewString()
-	for _, pod := range podList.Items[2:] {
-		expectedNames.Insert(pod.Name)
+	type podData struct {
+		podName  string
+		podPhase v1.PodPhase
 	}
 
-	var podPointers []*v1.Pod
-	for i := range podList.Items {
-		podPointers = append(podPointers, &podList.Items[i])
+	type test struct {
+		name          string
+		pods          []podData
+		expectedNames []string
```

Comment:
```suggestion
		wantNames []string
```
/v1/namespaces/default/po